tests/test_lightningd.py

```diff
@@ -94,6 +94,12 @@ def get_node(self, legacy=True):
             rpc = LegacyLightningRpc(socket_path, self.executor)
         else:
             daemon = utils.LightningD(lightning_dir, bitcoind.bitcoin_dir, port=port)
+            # If we have a disconnect string, dump it to a file for daemon.
+            if disconnect:
+                with open(os.path.join(lightning_dir, "dev_disconnect"), "w") as file:
+                    for d in disconnect:
```

Comment:
Nits: `file` is a keyword and discouraged for variable names, `f.write("\n".join(disconnect)` is a bit more idiomatic.